cabby/geo/walk.py

```diff
@@ -440,11 +447,63 @@ def get_points_and_route(map: map_structure.Map) -> Optional[item.RVSPath]:
                                                             main_pivot,
                                                             near_pivot,
                                                             beyond_pivot,
-                                                            cardinal_direction)
-
+                                                            cardinal_direction,
+                                                            number_intersections)
     return rvs_path_entity
 
 
+def get_number_interesctions_past(main_pivot: Dict, route: GeoDataFrame, map: map_structure.Map, end_point: Point) -> int:
+    '''Return the number of intersections between the main_pivot and goal. If the main_pivot and goal are on different streets return -1.
```

Comment:
Can remove "If the main_pivot and goal are on different streets return -1." here since it is clearly indicated in the "Returns" docstring below.